rhea/cores/fifo/fifo_fast.py

```diff
@@ -97,7 +97,10 @@ def rtl_fifo():
 
         elif fbus.write and not fbus.read:
             fbus.empty.next = False
-            if not fbus.empty:
+            # because addr starts from 0,
+            # there should not be an increment
+            # the first time 
+            if (not fbus.empty) and (not fbus.full):
```

Comment:
These need to be removed or verified that an SRL structured FIFO is created with the changes.  

If we remove the extra checks the tests can still verify that the expected exception is thrown when an overflow conditions occurs.

Did you get a response on the mailing-list?  I don't recall.  Either way, I don't think it affects this issue, an exception is still raise the exception can be verified as thrown or not thrown.  The setting of the value shouldn't matter because the execution (simulation) will stop because of the exception.